lib/review-apps.unit.test.js

```diff
@@ -1,39 +1,110 @@
 require('isomorphic-fetch');
 const nock = require('nock');
+const pRetry = require('p-retry');
 
-jest.mock('./github-api');
 const reviewApps = require('./review-apps');
+
 const BASE_HEROKU_API_URL = 'https://api.heroku.com';
 
+jest.mock('./github-api');
+
 describe('review-apps', () => {
+
 	let nockScope;
 	let reviewAppsNockScope;
+
 	beforeAll(() => {
+
 		nockScope = nock(BASE_HEROKU_API_URL)
 			.matchHeader('accept', 'application/vnd.heroku+json; version=3');
+
 		reviewAppsNockScope = nock(BASE_HEROKU_API_URL)
 			.matchHeader('accept', 'application/vnd.heroku+json; version=3.review-apps');
+
 		jest.mock('shellpromise', () => a => a, { virtual: true });
 
 		// WARNING: Disable error logs, to clean up test output
 		// Re-enable here if need be
 		jest.spyOn(console, 'error').mockImplementation().mockName('console.error');
+
 	});
 
 	afterAll(() => {
+
 		jest.unmock('shellpromise');
 		jest.unmock('./github-api');
+
 	});
 
 	afterEach(() => {
+
 		nock.cleanAll();
 		jest.resetAllMocks();
+
+	});
+
+	describe('runCheckForBuildSuccessStatus', () => {
+
+		const { runCheckForBuildSuccessStatus } = reviewApps;
+
+		it('returns without throwing an error if awaiting the review app build does not error', async () => {
+
+			const commit = 'a00000000000';
+			const appId = 'app123';
+
+			nockScope
+				.get('/apps/app123/builds')
+				.reply(200, [
+					{
+						status: 'succeeded',
+						app: {
+							id: appId
+						},
+						source_blob: {
+							version: commit
+						}
+					}
+				]);
+
+			try {
+				await runCheckForBuildSuccessStatus({ commit, appId, minTimeout: 0 });
+			} catch (error) {
+				expect(true).toBeFalsy();
+			}
+
+			expect(true).toBeTruthy();
+
+		});
+
+		it('throws pRetry.AbortError if the awaiting the review app build errors', async () => {
+
+			const commit = 'a00000000000';
+			const appId = 'app123';
+
+			nockScope
+				.get('/apps/app123/builds')
+				.times(60)
+				.reply(400, {
+					message: 'Some error occurred'
+				});
+
+			try {
+				await runCheckForBuildSuccessStatus({ commit, appId, minTimeout: 0 });
+			} catch (error) {
+				expect(error.message).toMatch('Some error occurred');
+				expect(error).toBeInstanceOf(pRetry.AbortError);
+			}
+
+		});
+
 	});
 
 	describe('createReviewApp', () => {
+
 		const { createReviewApp } = reviewApps;
 
 		it('returns created review app', async () => {
```

Comment:
I know you didn't write this test but i don't think its a good test in its current state.

It could be improved by doing the following:
- asserting the body of the request
- adding some data in the response to assert the full response is returned from `createReviewApp`

Reply:
> asserting the body of the request

Do you mean the body of the request made to `nock` (in this instance the POST request made to `/review-apps`)?

> adding some data in the response to assert the full response is returned from `createReviewApp`

👍

Reply:
> Do you mean the body of the request made to `nock`?

yes, that way we would know that the correct request is being made